src/event_imp.rs

```diff
@@ -1044,6 +1044,216 @@ fn test_debug_ready() {
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use mio::Interests;
+    ///
+    /// let interest = Interests::readable();
+    ///
+    /// assert!(interest.is_readable());
+    /// ```
+    ///
+    /// [`Poll`]: struct.Poll.html
+    #[inline]
+    pub fn readable() -> Interests {
+        Interests(READABLE)
+    }
+
+    /// Returns a `Interests` representing writable readiness.
+    ///
+    /// See [`Poll`] for more documentation on polling.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use mio::Interests;
+    ///
+    /// let interest = Interests::writable();
+    ///
+    /// assert!(interest.is_writable());
+    /// ```
+    ///
+    /// [`Poll`]: struct.Poll.html
+    #[inline]
+    pub fn writable() -> Interests {
+        Interests(WRITABLE)
+    }
+
+    /// Returns a `Interests` representing readable and writable readiness.
+    ///
+    /// See [`Poll`] for more documentation on polling.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use mio::Interests;
+    ///
+    /// let interest = Interests::both();
+    ///
+    /// assert!(interest.is_readable());
+    /// assert!(interest.is_writable());
+    /// ```
+    ///
+    /// [`Poll`]: struct.Poll.html
+    #[inline]
+    pub fn both() -> Interests {
+        Interests(READABLE | WRITABLE)
+    }
+    
+    /// Returns true if the value includes readable readiness
+    ///
+    /// See [`Poll`] for more documentation on polling.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use mio::Interests;
+    ///
+    /// let interest = Interests::readable();
+    ///
+    /// assert!(interest.is_readable());
+    /// ```
+    ///
+    /// [`Poll`]: struct.Poll.html
+    pub fn is_readable(&self) -> bool {
+        (self.0 & READABLE) != 0
+    }
+
+    /// Returns true if the value includes writable readiness
+    ///
+    /// See [`Poll`] for more documentation on polling.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use mio::Interests;
+    ///
+    /// let interest = Interests::writable();
+    ///
+    /// assert!(interest.is_writable());
+    /// ```
+    ///
+    /// [`Poll`]: struct.Poll.html
+    pub fn is_writable(&self) -> bool {
+        (self.0 & WRITABLE) != 0
+    }
+
+    /// Create a `Interests` instance using the given `usize` representation.
+    ///
+    /// The `usize` representation must be treated as opaque. There is no
+    /// guaranteed correlation between the returned value and platform defined
+    /// constants. Also, there is no guarantee that the `usize` representation
+    /// will remain constant across patch releases of Mio.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use mio::Interests;
+    ///
+    /// let interest = Interests::readable();
+    /// let interest_usize = interest.as_usize();
+    /// let interest2 = Interests::from_usize(interest_usize);
+    ///
+    /// assert_eq!(interest, interest2);
+    /// ```
+    pub fn from_usize(val: usize) -> Interests {
```

Comment:
Lets not make this public. Or do you have a use case for that? Same for `as_usize`.